.gitmodules

```diff
@@ -1,3 +1,6 @@
 [submodule "vcpkg"]
 	path = vcpkg
 	url = https://github.com/microsoft/vcpkg.git
+[submodule "traypp"]
```

Comment:
For what follows, I built a small demo...

https://github.com/lemire/lodepngdemo

It works portably (linux/windows) and it avoids entirely the issue of submodules.

Instead of creating a submodule for traypp, you can use https://github.com/cpm-cmake/CPM.cmake

create a cmake subdirectory in your project, and copy https://github.com/cpm-cmake/CPM.cmake/releases/latest/download/get_cpm.cmake into it, naming the file cmake/CPM.cmake.

See for example https://github.com/simdjson/simdjson/tree/master/cmake

Then you can do...

```
include(cmake/CPM.cmake)
```

in your main (root) CMakeLists.txt.

Then you can do...

```
  CPMAddPackage(
        NAME traypp
        URL https://github.com/Soundux/traypp/archive/698db7d58dd450cc9e30dc12d3bd0c5ca4d6a5b1.zip
  )
```
(where `698db7d58dd450cc9e30dc12d3bd0c5ca4d6a5b1` is the hash of the last commit.)

Right before `link_libraries(tray)`.

This has several benefits. E.g., if you need to patch the dependency, or point to another commit, it is trivial, no messing around with submodules.

Using CPM also allows you to grab libraries like `fmt` automatically...

```
CPMAddPackage("gh:fmtlib/fmt#11.0.2")
CPMAddPackage("gh:ocornut/imgui@1.91.2")
CPMAddPackage("gh:glfw/glfw#3.4")
```


You can also do...



```
  CPMAddPackage(
        NAME lodepng
        URL https://github.com/lvandeve/lodepng/archive/ed3c5f14b136c936d615ee3b38aaa7e7e642f12c.zip
  )
```


This avoids the need to pull the `lodepng` files in your repo, you can let CMake pull them for you (and this makes updates easy.)

Please see my demo.